packager/mpd/base/xml/xml_node.cc

```diff
@@ -429,6 +435,16 @@ bool RepresentationXmlNode::AddLiveOnlyInfo(
     if (IsTimelineConstantDuration(segment_infos, start_number)) {
       segment_template.SetIntegerAttribute("duration",
                                            segment_infos.front().duration);
+      if (FLAGS_dash_add_last_segment_number_when_needed) {
+        uint32_t last_segment_number = 0;
+        for (const auto& segment_info_element: segment_infos) {
```

Comment:
There should be a space between "segment_info_element" and ":". You may also use clang-format: https://github.com/google/shaka-packager/blob/master/packager/tools/git/check_formatting.py.